RecoTracker/TkDetLayers/src/Phase2OTBarrelRodBuilder.cc

```diff
@@ -5,58 +5,98 @@ using namespace edm;
-      innerGeomDetBrothers.push_back(theGeomDetBrother);
+    LogDebug("TkDetLayers") << "innerGeomDets.size(): " << innerGeomDets.size();
+    LogDebug("TkDetLayers") << "outerGeomDets.size(): " << outerGeomDets.size();
 
-    if (compGeometricDets[1]->positionBounds().perp() > meanRBrothers)
-      outerGeomDetBrothers.push_back(theGeomDetBrother);
-  }
+    return new Phase2OTBarrelRod(innerGeomDets, outerGeomDets, innerGeomDetBrothers, outerGeomDetBrothers);
+
+  } else {
+    vector<const GeometricDet*> compGeometricDets;
+
+    vector<const GeomDet*> innerGeomDetBrothers;
+    vector<const GeomDet*> outerGeomDetBrothers;
+
+    double meanRBrothers = 0;
+    for (vector<const GeometricDet*>::const_iterator it = allGeometricDets.begin(); it != allGeometricDets.end();
+         it++) {
+      compGeometricDets = (*it)->components();
+      if (compGeometricDets.size() != 2) {
+        LogDebug("TkDetLayers") << " Stack not with two components but with " << compGeometricDets.size() << std::endl;
+      } else {
+        //LogTrace("TkDetLayers") << " compGeometricDets[0]->positionBounds().perp() " << compGeometricDets[0]->positionBounds().perp() << std::endl;
+        //LogTrace("TkDetLayers") << " compGeometricDets[1]->positionBounds().perp() " << compGeometricDets[1]->positionBounds().perp() << std::endl;
+        meanR = meanR + compGeometricDets[0]->positionBounds().perp();
+        meanRBrothers = meanRBrothers + compGeometricDets[1]->positionBounds().perp();
+      }
+    }
+    meanR = meanR / allGeometricDets.size();
+    meanRBrothers = meanRBrothers / allGeometricDets.size();
+    LogDebug("TkDetLayers") << " meanR Lower " << meanR << std::endl;
+    LogDebug("TkDetLayers") << " meanR Upper " << meanRBrothers << std::endl;
 
-  LogDebug("TkDetLayers") << "innerGeomDets.size(): " << innerGeomDets.size();
-  LogDebug("TkDetLayers") << "outerGeomDets.size(): " << outerGeomDets.size();
-  LogDebug("TkDetLayers") << "innerGeomDetsBro.size(): " << innerGeomDetBrothers.size();
-  LogDebug("TkDetLayers") << "outerGeomDetsBro.size(): " << outerGeomDetBrothers.size();
+    for (vector<const GeometricDet*>::iterator it = allGeometricDets.begin(); it != allGeometricDets.end(); it++) {
+      compGeometricDets = (*it)->components();
+      const GeomDet* theGeomDet = theGeomDetGeometry->idToDet(compGeometricDets[0]->geographicalID());
+      LogTrace("TkDetLayers") << " inserisco " << compGeometricDets[0]->geographicalID().rawId() << std::endl;
+
+      if (compGeometricDets[0]->positionBounds().perp() < meanR)
+        innerGeomDets.push_back(theGeomDet);
+
+      if (compGeometricDets[0]->positionBounds().perp() > meanR)
+        outerGeomDets.push_back(theGeomDet);
+
+      const GeomDet* theGeomDetBrother = theGeomDetGeometry->idToDet(compGeometricDets[1]->geographicalID());
+      LogTrace("TkDetLayers") << " inserisco " << compGeometricDets[1]->geographicalID().rawId() << std::endl;
```

Comment:
@JanFSchulte please change it here as well

Reply:
Done.